### ISSUE TITLE
Question on `sqrt(max(var, eps))`

### ISSUE THREAD

**Issue by nmhkahn** [vacancy/Synchronized-BatchNorm-PyTorch#38]
Hi, really appreciate the great code.

I just got one question on computing the invstd ([#L150](https://github.com/vacancy/Synchronized-BatchNorm-PyTorch/blob/master/sync_batchnorm/batchnorm.py#L150)), which uses `sqrt(max(var, eps))` instead of `sqrt(var+eps)`.
For example, in [Normalization.cuh in pytorch](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/Normalization.cuh#L198), it seems that they compute invstd with `sqrt(var+eps)` as below.

```c++
template<typename T>
struct InvStd {
  __device__ __forceinline__ T operator()(T var, double epsilon) const {
    T invstd = 0;
    if (var != static_cast<T>(0) || epsilon != static_cast<T>(0)) {
      invstd = static_cast<T>(1) / device_sqrt(var + epsilon);
    }
    return invstd;
  }
};
```
or is there any reason to choose the way of calculating invstd as now?
Thanks!

**Comment by vacancy** [comment]
That's a really good question. I think I have tested both. My test result was (based on a very old pytorch implementation ~0.3 I think),
using max(var, eps) outputs closer results (the output or the gradient) to the PyTorch's batchnorm implementation than var + eps.

Maybe it's time to do some new tests.

**Comment by nmhkahn** [comment]
Thanks for sharing your experience.
I observed that switching to sqrt(var+eps) results in slightly better performance on some image-to-image translation methods. But not 100% sure since I didn't fix the random seed nor deeply dug in the result.

Close this issue but let me know if you get some new results 👍
